dotnet-desktop-guide/net/wpf/documents/snippets/how-to-display-print-dialog/csharp/MainWindow.xaml.cs

```diff
@@ -0,0 +1,227 @@
+﻿using System;
+using System.IO;
+using System.Reflection;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Documents;
+using System.Windows.Media;
+using System.Windows.Xps.Packaging;
+using Microsoft.Win32;
+
+namespace CodeSampleCsharp
+{
+    /// <summary>
+    /// Interaction logic for MainWindow.xaml
+    /// </summary>
+    public partial class MainWindow : Window
+    {
+        public MainWindow()
+        {
+            InitializeComponent();
+
+            // Events
+            btnBrowse.Click += BtnBrowse_Click;
+            btnPrint.Click += BtnPrint_Click;
+        }
+
+        /// <summary>
+        /// Browse button click event handler
+        /// Gets the path to a source XPS file
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
```

Comment:
For reference, we dont need these in the snippets code. The way we have this repo designed, we don't use snippets as samples. They're purely for quoting in the article. Code comments in the snippet are obviously helpful so those are good to add.

Reply:
Good to know. I'll not spend time commenting a non-snippet method unless the code is unobvious.